### ISSUE TITLE
Titelbild verkleinern

### ISSUE THREAD

**Issue by innosmith** [decidim-swiss/decidim.swiss#52]
Die Kundin wünscht auf der Startseite schneller die Inhalte unterhalb des Titelbildes sichtbar zu machen. Hierzu soll das Titelbild um 25-33% in der Höhe reduziert werden. Ideal wäre die Grösse des Titelbildes von MeinQuartier.Zuerich. Allenfalls kann zusätzlich noch die oberste Navigationssection etwas verkleinert werden.

Fragen zur Lösung: 
- Ist es korrekt, dass aufgrund der ursprünglichen Slider-Lösung das Titelbild nicht einfach von Non-Devs angepasst werden kann?
- Wie aufwändig wäre es, dies so umzubauen, dass wir das Hintergrundbild hinter dem Logo einfach selber anpassen können?

**Comment by carlobeltrame** [comment]
Die Frage bei "selber anpassen" ist wie immer, wo wird das Bild hochgeladen und gehostet. Da die Formulierung des Issues relativ offen ist und verschiedene Wünsche erwähnt, sehe ich viele Umsetzungsmöglichkeiten. Ich versuche diese mal aufzulisten, sortiert von wenig bis viel Aufwand.

1. Decidim bietet einen Content Block "Hauptbild" den ihr einfach auf die Startseite ziehen und selber ein Bild hochladen könnt. Gestaltungsmöglichkeiten haben wir nur via CSS, also können Farben, Abstände, Schriftgrössen und Schriftarten geändert werden und Elemente entfernt werden, aber keine neuen Elemente hinzugefügt werden.
**Bilder selber hochladen:** ja
**Höhe bestimmen:** ja, mit CSS
**Deinklima-Logo übers Bild legen:** nein
**Erweiterbar sodass irgendwann wieder das Karussell verwendet werden könnte:** nein
**Vorteile:** Wenig bis kein Umsetzungsaufwand, intuitives UI um das Bild zu ersetzen
**Nachteile:** Deinklima-Logo kann nicht über das Bild gelegt werden, also müsste das Logo weiterhin direkt auf dem Hintergrundbild drauf sein.

2. Lösung wie bisher. Bild wird wie bei meinquartier.zuerich irgendwo in Decidim hochgeladen (z.B. in einer privaten Organisation oder wie bei Luzern im "Hauptbild"-Block bei den Startseite-Einstellungen) und der korrekte Link (achtung, es gibt jeweils zwei Links, nur einer davon funktioniert dauerhaft) wird im HTML-Block eingetragen.
**Bilder selber hochladen:** ja
**Höhe bestimmen:** ja, mit CSS
**Deinklima-Logo übers Bild legen:** ja, mit einmaligem Entwicklungsaufwand von ca. 0.5 - 1PT
**Erweiterbar sodass irgendwann wieder das Karussell verwendet werden könnte:** je nach Speicherort, aber macht es noch intransparenter / fehleranfälliger für Plattform-Admins
**Vorteile:** Wenig bis kein Umsetzungsaufwand
**Nachteile:** Das Quell-Bild oder die private Organisation etc. könnten aus versehen gelöscht werden, weil der Bezug zur Startseite nicht intuitiv klar ist. Wenn das Bild geändert wird, muss auch daran gedacht werden, den Link zum neuen Bild im HTML-Block zu aktualisieren.

3. Lösung wie bisher. Bild wird auf einem externen Hosting-Service hochgeladen (z.B. vormaliges Metanet-Hosting) und der Link wird im HTML-Block eingetragen.
**Bilder selber hochladen:** ja, aber braucht separaten Zugang zum Hosting
**Höhe bestimmen:** ja, mit CSS
**Deinklima-Logo übers Bild legen:** ja, mit einmaligem Entwicklungsaufwand von ca. 0.5 - 1PT
**Erweiterbar sodass irgendwann wieder das Karussell verwendet werden könnte:** ja
**Vorteile:** Wenig bis kein Umsetzungsaufwand
**Nachteile:** Zusätzlicher Zugang für Hosting ist nötig um das Bild zu ändern. Verantwortung und Finanzierung des Hostings müsste geklärt werden. Wenn das Hosting mal nicht oder langsam reagiert, wird auch kein Bild angezeigt (vgl. kürzliche Abklärungen bei meinquartier.zuerich zu langer Karten-Ladezeit von externen Servers).

4. Zusätzlichen Datei-Upload irgendwo in Decidim einprogrammieren (z.B. im System Dashboard oder bei den Einstellungen im Admin Dashboard). Der fix von Decidim generierte Link wird im HTML-Block eingetragen.
**Bilder selber hochladen:** ja
**Höhe bestimmen:** ja, mit CSS
**Deinklima-Logo übers Bild legen:** ja, mit einmaligem Entwicklungsaufwand von ca. 0.5 - 1PT
**Erweiterbar sodass irgendwann wieder das Karussell verwendet werden könnte:** ja, aber mit nochmals mehr Entwicklungsaufwand (+ 1-2PT)
**Vorteile:** Intuitives UI um das Bild zu ersetzen
**Nachteile:** Mit Abstand höchster Umsetzungsaufwand (ca. 3PT, genaue Umsetzung noch unklar)

Wie du siehst ist "Titelbild verkleinern" in jedem Fall das geringste Problem der gewünschten Änderungen.

**Comment by innosmith** [comment]
Danke für deine Analyse @carlobeltrame. Aktuell ist das Logo noch Teil des Bildes (richtig?), entsprechend könnte ich/wir es mit Variante 1 umsetzen. Wenn wir später im Kontext des geplanten Re-Design Prozesses für deinklima.ch die Anforderung festlegen, dass das Logo als eigener Layer über das Bild gefügt werden soll, können wir eine der Varianten 2-4 wählen. Macht das für dich auch Sinn?

**Comment by carlobeltrame** [comment]
Ja, genau, bisher ist es Teil des Bilds (anders als bei meinquartier.zuerich). Klingt gut für mich. Melde dich falls du Hilfe mit CSS brauchst.

**Comment by innosmith** [comment]
Variante 1 mit der "Hauptbild" Komponente von Decidim auf der Startseite ist keine hübsche Lösung, habe es getestet. Je nach Bildschirmbreite und Bild wiederholt sich dieses und das Logo ist bekanntlich statisch. Zusätzlich wird ein Text sowie ein Call2Action Button angezeigt. Letzteres könnte man wahrscheinlich mit CSS ausblenden.

Gewünscht ist Variante 2 analog Dialog Luzern oder MeinQuartier, sodass die Kundin selbst das Hintergrundbild bei Bedarf einfach wechseln kann auf der Decidim-Plattform. Die Karussell-Komponenten gerne entfernen. Es ist noch zu klären, welches die einfachste Variante als Speicherort für das Startbild ist. Als Datei einer privaten Organisation dürfte unpassend sein, da Decidim private Ressourcen eigentlich nicht für Unberechtigte teilen dürfte.

Wo besteht noch Unsicherheit @carlobeltrame in der Aufwandschätzung von 4-8h, wenn wir 1:1 Dialog Luzern oder MeinQuartier kopieren?

**Comment by carlobeltrame** [comment]
Unsicherheit bzw. mögliche Unterschiede zu Dialog Luzern und MeinQuartier.zuerich bestehen im Bildformat in welchem das Logo zu Verfügung gestellt wird, in der gewünschten Anzeigegrösse des Logos, sowie bei Edge Cases auf kleinen Bildschirmen, wo die Seitenverhältnisse und Grösse des Logos eine Rolle spielen. Bei Dialog Luzern wurde zudem glaube ich noch ein kaum aktiv bemerkbarer Schlagschatten für besseren Kontrast zum Hintergrundbild eingefügt.

**Comment by innosmith** [comment]
Danke für die Rückmeldung. D.h. wenn sich z.B. das Logo ändert, könnte es auch wieder zu kleinen Anpassungsaufwänden führen. Welche Detailvariante MeinQuartier vs. Dialog Luzern findest du aus technischer sowie wirtschaftlicher Sicht geeigneter?

**Comment by larsUE** [comment]
Close hier weil dies eine Pre-Redesign Funktionseinschränkung betrifft, die nun nicht mehr auftritt.
